### PR TITLE
improve(kvstore_instance_classes): add validateFunc for engine

### PR DESCRIPTION
**Pull request by handsomepotato**
image
image

### CODE REVIEW

File: alicloud/data_source_alicloud_kvstore_instance_classes.go [hunk 1]
```diff
@@ -26,10 +26,11 @@ func dataSourceAlicloudKVStoreInstanceClasses() *schema.Resource {
...
 			},
 			"engine": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Default:  "redis",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"redis", "memcache"}, true),
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_kvstore_instance_classes.go]: Please keep sync with resource alicloud_kvstore_instance validatefunc

File: alicloud/data_source_alicloud_kvstore_instance_engines.go [hunk 2]
```diff
@@ -21,10 +21,11 @@ func dataSourceAlicloudKVStoreInstanceEngines() *schema.Resource {
...
 			},
 			"engine": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Default:  "redis",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"redis", "memcache"}, true),
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_kvstore_instance_engines.go]: still applies

File: alicloud/data_source_alicloud_kvstore_instance_classes.go (hunk 1 above)
Comment (handsomepotato) [reply alicloud/data_source_alicloud_kvstore_instance_classes.go]: done

File: alicloud/data_source_alicloud_kvstore_instance_engines.go (hunk 2 above)
Comment (handsomepotato) [reply alicloud/data_source_alicloud_kvstore_instance_engines.go]: done

**Pull request merged**